OkHttpClientTransportTest.pendingStreamFailedByIdExhausted is flaky

As seen in the [Travis logs](https://s3.amazonaws.com/archive.travis-ci.org/jobs/60558899/log.txt
):
```
io.grpc.transport.okhttp.OkHttpClientTransportTest > pendingStreamFailedByIdExhausted FAILED
    java.lang.AssertionError: Failed waiting stream to be closed.
        at org.junit.Assert.fail(Assert.java:88)
        at io.grpc.transport.okhttp.OkHttpClientTransportTest$MockStreamListener.waitUntilStreamClosed(OkHttpClientTransportTest.java:912)
        at io.grpc.transport.okhttp.OkHttpClientTransportTest.pendingStreamFailedByIdExhausted(OkHttpClientTransportTest.java:650)

    java.lang.AssertionError: expected:<0> but was:<1>
        at org.junit.Assert.fail(Assert.java:88)
        at org.junit.Assert.failNotEquals(Assert.java:743)
        at org.junit.Assert.assertEquals(Assert.java:118)
        at org.junit.Assert.assertEquals(Assert.java:555)
        at org.junit.Assert.assertEquals(Assert.java:542)
        at io.grpc.transport.okhttp.OkHttpClientTransportTest.tearDown(OkHttpClientTransportTest.java:142)
```

The commit was restarted, and [succeeded](https://travis-ci.org/grpc/grpc-java/builds/60558898).

Another example of apparent flakiness: https://travis-ci.org/grpc/grpc-java/builds/59916295 . The commit before and after were green. The tested commit only changed Netty code.

--- a/okhttp/src/test/java/io/grpc/transport/okhttp/OkHttpClientTransportTest.java
+++ b/okhttp/src/test/java/io/grpc/transport/okhttp/OkHttpClientTransportTest.java
@@ -627,6 +627,8 @@ public class OkHttpClientTransportTest {
         newStreamReturn2.countDown();
       }
     }).start();
+    // Wait until stream2 is pending, to make sure stream2 is queued in front of stream3.
+    waitForStreamPending(1);
     new Thread(new Runnable() {
       @Override
       public void run() {

